From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Wed, 5 Aug 2026 11:25:50 +0200
Subject: [PATCH 36/67] Add and adjust embedlite static prefs

Generate the EmbedLite APZ preference group required by hosted remote content.

Restore the external-protocol permission gate with a secure default.
---
 modules/libpref/init/StaticPrefList.yaml      | 22 +++++++++++++++++++
 modules/libpref/moz.build                     |  1 +
 .../handling/ContentDispatchChooser.sys.mjs   | 10 +++++++++
 3 files changed, 33 insertions(+)

diff --git a/modules/libpref/init/StaticPrefList.yaml b/modules/libpref/init/StaticPrefList.yaml
index 722d74c6ebef..4b970f52ff0b 100644
--- a/modules/libpref/init/StaticPrefList.yaml
+++ b/modules/libpref/init/StaticPrefList.yaml
@@ -6476,6 +6476,28 @@
 #endif
   mirror: always
 
+#ifdef MOZ_EMBEDLITE
+#---------------------------------------------------------------------------
+# Prefs starting with "embedlite."
+#---------------------------------------------------------------------------
+
+- name: embedlite.azpc.handle.viewport
+  type: bool
+  value: true
+  mirror: always
+
+- name: embedlite.azpc.handle.singletap
+  type: bool
+  value: true
+  mirror: always
+
+- name: embedlite.azpc.handle.longtap
+  type: bool
+  value: true
+  mirror: always
+
+#endif
+
 #---------------------------------------------------------------------------
 # Prefs starting with "extensions."
 #---------------------------------------------------------------------------
diff --git a/modules/libpref/moz.build b/modules/libpref/moz.build
index fa05921295e8..367f2e841277 100644
--- a/modules/libpref/moz.build
+++ b/modules/libpref/moz.build
@@ -42,6 +42,7 @@ pref_groups = [
     "docshell",
     "dom",
     "editor",
+    "embedlite",
     "extensions",
     "fission",
     "font",
diff --git a/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs b/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs
index 8d89744fc8d9..21be36db7f43 100644
--- a/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs
+++ b/toolkit/mozapps/handling/ContentDispatchChooser.sys.mjs
@@ -20,6 +20,12 @@ XPCOMUtils.defineLazyPreferenceGetter(
   "network.protocol-handler.prompt-from-external",
   true
 );
+XPCOMUtils.defineLazyPreferenceGetter(
+  gPrefs,
+  "requireExternalProtocolPermission",
+  "security.external_protocol_requires_permission",
+  true
+);
 
 const PROTOCOL_HANDLER_OPEN_PERM_KEY = "open-protocol-handler";
 const PERMISSION_KEY_DELIMITER = "^";
@@ -290,6 +296,10 @@ export class nsContentDispatchChooser {
     aTriggeredExternally,
     isStandardProtocol
   ) {
+    if (!gPrefs.requireExternalProtocolPermission) {
+      return true;
+    }
+
     // If a handler is set to open externally by default we skip the dialog.
     const { type, hasDefaultHandler, preferredApplicationHandler } = aHandler;
 
